### ISSUE TITLE
Introduce new CSS variables: --hh-fixed-header-height and --hh-fixed-footer-height

### ISSUE THREAD

**Issue by marc-farre** [humhub/humhub#7131]
### Is your feature request related to a problem? Please describe.

In some situations, we need to know the "Available Content Area" which is:
- the "Viewport Height"
- minus the "Fixed Header Height"
- minus the "Fixed Footer Height"

Usage examples:
- The Messenger module (see https://github.com/humhub/mail/issues/406)
- The Members Map module
- The Events Map module
- And certainly other usages

Currently, we need to get the body padding top for the HumHub community theme.

Problems:
- This value is not available with CSS (we need JavaScript)
- The "Available Content Area" must be calculated differently for other themes (e.g., the Clean Theme has a bottom menu on mobiles)

### Describe the solution you'd like

Create 2 new CSS variables:
- `--fixed-header-height`
- `--fixed-footer-height`

The values can be changed dynamically:
- With CSS:
  ```css
    @media (min-width: 768px) {
      :root {
        --fixed-header-height: 130px;
      }
    }
  ```
- With JS:
  ```js
    $(':root').css('--fixed-header-height', '130px');
  ```

Then we can get the "Available Content Area" in CSS using the `calc()` function:
```css
height: calc(100vh - var(--fixed-header-height) - var(--fixed-footer-height));
```

### Describe alternatives you've considered

Create a `--available-content-variable` but:
- we won't have the `var(--fixed-header-height)` which could be used, e.g. to set the body padding top
- it's less explicit than the 2 other variable suggestions

**Comment by luke-** [comment]
@marc-farre For me such a CSS variable sounds very useful and avoid hacks in different places.

Do you see also any problems/cons, when we always use such a var:

I am thinking of things like 

- an additional line due to dynamic content
- Embedded Accessibility Tools
- or other CSS adjustments 

Maybe we could also update it via JS?


@yurabakhtin What do you think about?

**Comment by luke-** [comment]
@marc-farre 

> @luke- Sorry, I don't understand "an additional line due to dynamic content".

e.g. In earlier versions there was a bug when the main navigation had too many entries, there was a line break.

But currently I can't see any reason why the calculated header height could be increased by content.

**Comment by marc-farre** [comment]
@luke- @yurabakhtin PR https://github.com/humhub/humhub/pull/7133
And PR https://github.com/humhub/enterprise-theme/pull/103 for the Enterprise Theme

I've added a `hh-` prefix to the `--hh-fixed-header-height` and `--hh-fixed-footer-height` CSS variables.

I think we should progressively prefix all CSS variables to avoid conflicts and to follow BS5 recommendations: https://getbootstrap.com/docs/5.3/customize/css-variables/

- Bootstrap prefix: `bs-`
- HumHub specific variables prefix: `hh-`
- Custom theme prefix, e.g. Enterprise Theme: `et-`? Or `hh-et-`?

**Comment by luke-** [comment]
@marc-farre 

> I think we should progressively prefix all CSS variables to avoid conflicts and to follow BS5 recommendations: https://getbootstrap.com/docs/5.3/customize/css-variables/
> 
> * Bootstrap prefix: `bs-`
> * HumHub specific variables prefix: `hh-`
> * Custom theme prefix, e.g. Enterprise Theme: `et-`? Or `hh-et-`?

Looks good. For Enterprise Theme I would prefer `hh-et-`

**Comment by marc-farre** [comment]
@yurabakhtin An idea for the Enterprise Theme issue (and other use cases):

Instead of having fixed elements with `position: absolute;` and adding padding to the parent element (which causes problems when the height of the fixed element can change), why not using this CSS:

https://codepen.io/marc-farre/pen/jOjqMgJ

**Comment by yurabakhtin** [comment]
@marc-farre I think we should apply the same styles with scroller as we have on the small screens for the Space menu.
The list `<ul class="nav navbar-nav visible-md visible-lg">` should has styles like `display: flex !important; overflow-x: auto;`.

**Comment by marc-farre** [comment]
@yurabakhtin With your suggestion, it looks like this:

![image](https://github.com/user-attachments/assets/024dc48a-f75f-41b5-8aad-cbbefd57f15f)

I've opened an issue to discuss it because I think it's not related to this one: https://github.com/humhub/humhub/issues/7135

**Comment by luke-** [comment]
@marc-farre What is the status here?

**Comment by marc-farre** [comment]
@luke- 

- PR https://github.com/humhub/humhub/pull/7133 is merged
- PR https://github.com/humhub/enterprise-theme/pull/103 should be merged
- @yurabakhtin suggestion is about another subject for which I opened this issue: https://github.com/humhub/humhub/issues/7135

So I think that once the PR for the Enterprise theme is merged, we can close this issue.

**Comment by luke-** [comment]
@marc-farre Ok great! I'll close this. We'll merge the V17 PRs after Beta release.

**Comment by marc-farre** [comment]
@luke- I've been warned by @mbumpalumpa about an issue with the Banner module when using the Enterprise Theme.

This due to the unmerged PR https://github.com/humhub/enterprise-theme/pull/103

I have rebased the PR from master and enhance it:
- https://github.com/humhub/enterprise-theme/pull/103/commits/5caf98e7fc6f9d29668d01e1a3f988d37016f00e
- https://github.com/humhub/enterprise-theme/pull/103/commits/818f8c229912a4d12af0231e4068076de4806549

In https://github.com/humhub/enterprise-theme/pull/103/files, see changes in:
- the Changelog
- the `main.php` view where I add the `enterprise-theme` class

This class will allow me to add a specific CSS to the Space menu when the banner is enabled (and can be used for other purposes later too). I've already done the same thing with the Clean Theme.

**Comment by marc-farre** [comment]
PS: @luke- we have an error with self tests when trying to "Create a space category". I don't see how it can be related to this PR, but do you want me to fix it?

**Comment by marc-farre** [comment]
@luke- I took a closer look at the failing test.

When the self test click here:

![Image](https://github.com/user-attachments/assets/39455bb0-379a-4385-9b16-a43afda76791)

Instead of having this page:

![Image](https://github.com/user-attachments/assets/176dba12-2608-4958-88ae-a2550cf02ad0)

We have this:

![Image](https://github.com/user-attachments/assets/6da0c52b-9db0-4994-a23a-b03f38b41b78)

On my local tests, I don't have any issue.

But it could be explained with `DeviceDetectorHelper::getBodyClasses()` that has be introduced recently for 1.17.

Is there a way to get the PHP error logs and to know what version of HumHub is installed on GitHub?

**Comment by luke-** [comment]
@yurabakhtin Can you please add the upload of logs also to the module tests?

**Comment by luke-** [comment]
@marc-farre Can you please try again?

**Comment by marc-farre** [comment]
@luke- @yurabakhtin The issue was on the develop branch (I didn't pay attention on that, and I was testing on master), because of the old `DataSaved()` widget which is removed on develop.

Commit https://github.com/humhub/enterprise-theme/pull/103/commits/803962f38ef7e7973ff015bd69bbf1cda04c47cf fixes the issue (and tests).

PR https://github.com/humhub/enterprise-theme/pull/103 is now ready.

**Comment by yurabakhtin** [comment]
@marc-farre @luke- To remove the deprecated widget we have the not merged PR https://github.com/humhub/enterprise-theme/pull/130. But it seems I was wrong there when I set `minVersion` to `1.18`, I think the PR can work without errors on `1.17` as well.

**Comment by luke-** [comment]
@marc-farre If https://github.com/humhub/enterprise-theme/pull/130/files needs v18, I would wait until then. Possibly merge it into the `develop` or `bs5` branch.

**Comment by marc-farre** [comment]
@luke- it doesn't need 1.18. It's for 1.17.0-beta.1 and later.

**Comment by luke-** [comment]
@marc-farre Thanks! Just release ET 1.14.5
